File: 프로그래머스/lv2/131127. 할인 행사/할인 행사.py
from collections import Counter
def solution(want, number, discount):
    answer = 0

    for i in range(len(discount)):
        cnt = Counter(discount[i:i+10])
        temp = 0
        for j in range(len(want)):
            if cnt[want[j]] == number[j]:
                temp += 1

        if temp == len(want):
            answer += 1

    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.java
import java.util.*;
class Solution {
    public int solution(String[] want, int[] number, String[] discount) {
        int answer = 0;
        for (int i = 0; i < discount.length - 9; i++) {
            if (check(i, want, number, discount)) answer++;
        }
        return answer;
    }

    private boolean check(int start, String[] want, int[] number, String[] discount){
        Map<String, Integer> map = new HashMap<>();
        for (int i = 0; i< want.length; i++) {
            map.put(want[i], number[i]);
        }
        for (int i = start; i < start + 10; i++) {
            int value = map.getOrDefault(discount[i], -1);
            if (value != -1) {
                map.put(discount[i], value-1);
            }
        }
        for (int i= 0; i < want.length; i++) {
            int value = map.get(want[i]);
            if (value > 0) return false;
        }
    
        return true;
    }
}